smt/utils/design_space.py

```diff
@@ -927,11 +934,18 @@ def _sample_valid_x(
         """Sample design vectors"""
         # Simplified implementation: sample design vectors in unfolded space
         x_limits_unfolded = self.get_unfolded_num_bounds()
-        if self.seed is None:
-            self.seed = random_state
+        if self.random_state is None:
+            self.random_state = random_state
 
         if self._cs is not None:
             # Sample Configuration objects
+            if not (hasattr(self, "seed")):
```

Comment:
Testing the presence of an attribute is brittle. We should discuss to see if we can avoid that.